lib/forspell/loaders/c.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+require_relative 'ruby'
+
+module Forspell::Loaders
+  class C < Ruby
+    def initialize(file: nil)
+      super
+      @parser_class = YARD::Parser::C::CParser
+    end
+
+    def load_comments
+      @input = @input.encode('UTF-8', invalid: :replace, replace: '?') unless @input.valid_encoding?
```

Comment:
It is better to reload parent's `input` class, the goal is "this particular class has some input post-processing".